src/Microsoft.Azure.WebJobs.ServiceBus/ServiceBusHostBuilderExtensions.cs

```diff
@@ -14,13 +15,12 @@ public static class ServiceBusHostBuilderExtensions
     {
         public static IWebJobsBuilder AddServiceBus(this IWebJobsBuilder builder)
         {
-            builder.AddExtension<ServiceBusExtensionConfig>();
-
-            builder.Services.AddOptions<ServiceBusOptions>()
-                            .Configure<IConnectionStringProvider>((o, p) =>
-                            {
-                                o.ConnectionString = p.GetConnectionString(ConnectionStringNames.ServiceBus);
-                            });
+            builder.AddExtension<ServiceBusExtensionConfig>()
+                .ConfigureOptions<ServiceBusOptions>((config, options) =>
+                {
+                    config.Bind(options);
+                    options.ConnectionString = config.GetConnectionString(ConnectionStringNames.ServiceBus);
```

Comment:
This is an example of using `ConfigureOptions` (instead of `BindOptions`), which will give you the entire configuration section so you can assign/manager configuration as needed.